builtin/string.mbt

```diff
@@ -31,3 +65,45 @@ pub fn substring(
   guard start >= 0 && start <= end && end <= self.length()
   unsafe_substring(self, start, end)
```

Comment:
in the future, for patterns like this, `guard` followed by a function call, we may always in the function to eliminate the guard statement cc @Yu-zh